tests/unit/test_client.py

```diff
@@ -219,6 +219,32 @@ def test_ctor_w_client_info(self):
         self.assertEqual(list(client._batch_stack), [])
         self.assertIs(client._connection._client_info, client_info)
 
+    def test_ctor_w_emulator_w_env_var_glcoud_project(self):
+        from google.cloud.storage._helpers import STORAGE_EMULATOR_ENV_VAR
+
+        HOST = "https://api.example.com"
+        PROJECT = "test"
+
+        with mock.patch("os.environ", {STORAGE_EMULATOR_ENV_VAR: HOST}):
+            with mock.patch("os.getenv", side_effect=[HOST, PROJECT]):
+                client = self._make_one()
+                self.assertEqual(client.project, PROJECT)
+                self.assertEqual(client._connection.API_BASE_URL, HOST)
```

Comment:
This test needs an assertion that the client's credentials is an instance of `AnonymousCredentials`.

Also, rather than patching `os.getenv`, this would be clearer if we just patched in the expected environment:
```suggestion

    def test_ctor_w_emulator_w_env_var_glcoud_project(self):
        from google.auth.credentials import AnonymousCredentials
        from google.cloud.storage._helpers import STORAGE_EMULATOR_ENV_VAR

        HOST = "https://api.example.com"
        PROJECT = "test"
        environ = {
            STORAGE_EMULATOR_ENV_VAR: HOST,
            "GCLOUD_PROJECT": PROJECT,
        }

        with mock.patch("os.environ", environ):
            client = self._make_one()

        self.assertEqual(client.project, PROJECT)
        self.assertEqual(client._connection.API_BASE_URL, HOST)
        self.assertIsInstance(client._credentials, AnonymousCredentials)
```